### PR TITLE
[ops] - authentication

### PR DESCRIPTION
**Pull request by J6K**
- middleware to create tokens and validate users
- rooms are now slings
- client can only create a sling if they possess a token from login

- make sure to add TOKEN_SECRET into .env with whatever string you need
- or just build the env file again

### CODE REVIEW

**Comment by obber** [comment]
Can you check the build failure and fix up the errors? 

https://travis-ci.org/codesling/codesling.io/builds/299897814?utm_source=github_status&utm_medium=notification

Looks like one linter and a test error. Feel free to update the snapshot tests if necessary.

File: client/src/components/LandingPage/index.js [hunk 1]
```diff
@@ -9,14 +9,25 @@ import './LandingPage.css';
...
+    if(localStorage.token) {
+      this.setState({ authenticated: true });
+    }
+  }
+
   fetchSlingId = async () => {
     try {
-      const resp = await axios.get(`${process.env.REACT_APP_REST_SERVER_URL}/api/new-sling`);
-      const { slingId } = resp.data;
+      const { data } = await axios.get(`${process.env.REACT_APP_REST_SERVER_URL}/api/new-sling`,{
```
Comment (J6K) [inline client/src/components/LandingPage/index.js]: Can we always de-structure whenever possible?

File: rest-server/tests/newSling.test.js [hunk 2]
```diff
@@ -1,15 +1,22 @@
 import axios from 'axios';
 import dotenv from 'dotenv';
```
Comment (obber) [inline rest-server/tests/newSling.test.js]: The fix for the test case not working:

add an import for the http adapter under axios:

```javascript
import http from 'axios/lib/adapters/http';
```

File: rest-server/tests/newSling.test.js [hunk 3]
```diff
@@ -1,15 +1,22 @@
...
 test('rest-server should respond with a 200 to {GET /api/new-sling}', async () => {
   let resp = {};
+  const { accessToken } = generateToken({username: 'test', id: 1 });
   try {
-    resp = await axios.get(URL + '/api/new-sling');
+    resp = await axios.get(URL + '/api/new-sling', {
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      }
+    });
```
Comment (obber) [inline rest-server/tests/newSling.test.js]: Then add the adapter here:

```diff
resp = await axios.get(URL + '/api/new-sling', {
  headers: {
+   adapter: http
    Authorization: `Bearer ${accessToken}`,
  }
});
```

**Pull request merged**